skip-to/skip-to.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Plugin Name: Skip-to for classic themes
+ * Description: This is a copy of the block theme skip-to functionality for classic themes.
+ * 
+ * See: https://github.com/WordPress/wordpress-develop/blob/f7d2a2ee9d003633e6c729c0835cd4addd23f9b3/src/wp-includes/theme-templates.php#L101-L205
+ */
+namespace WordPressdotorg\MU_Plugins\Skip_To_Links {
+	function skip_to( $selector = 'main', $css = false ) {
+
+		$selector = apply_filters( 'wporg_skip_link_target', $selector );
+		if ( ! $selector ) {
+			return;
+		}
+
+		// Noop for Block themes. Use the_block_template_skip_link() instead.
+		if (
+			'main' === $selector &&
+			current_theme_supports( 'block-templates' ) &&
+			! empty( $_wp_current_template_content )
+		) {
+			return;
+		}
+
+		if ( $css ) {
+			add_action( 'wp_head', __NAMESPACE__ . '\css' );
+		}
+
+		add_action( 'wp_body_open', function() use( $selector ) {
+			skip_tag( $selector );
+		}, -1 );
+
+		// If a HTML ID is not passed, some JS will be needed.
+		if ( '#' !== substr( $selector, 0, 1 ) ) {
+			add_action( 'wp_footer', __NAMESPACE__ . '\js' );
+		}
+
+		// Don't allow a second call to this functionality.
+		add_filter( 'wporg_skip_link_target', '__return_false', 100 );
+	}
+
+	/**
+	 * Print the skip link.
+	 */
+	function skip_tag( $selector ) {
+		$target = ( '#' === substr( $selector, 0, 1 ) ? $selector : '' );
+		$tabindex = ( $target ? '' : 'tabindex="-1"' ); // Will be removed once the target is set.
+		printf(
+			'<a id="wporg-skip-link" %s class="skip-link screen-reader-text" href="%s" data-selector="%s">%s</a>'."\n",
+			$tabindex,
+			esc_attr( $target ),
+			esc_attr( $selector ),
+			__( 'Skip to content', 'wporg' )
```

Comment:
Since the `selector` can be customized, is there ever a case where this might not skip to content, but to another section? Or there might be more than one skip on a page? If so, this text should also be a parameter so it can be controlled.

Reply:
The customization of the skip destination was primarily to avoid having to change all of the existing themes structure, while removing the header dependancies.

I considered that, and decided against supporting multiple skip-links per page, for this at least. Similarly, it looks like the Core skip functionality also doesn't support multiple skiplinks, only linking to the singular individual `<main>` element.

In https://github.com/WordPress/wordpress.org/pull/49/files#diff-b468acd1a227cd79ee142be67224a169761ccddb47f283c5d384523d67fb0c1a you can see an example of where there were previously two skip to links, that I've condensed to one.

In that case, the links didn't make sense as they were.
 - Start of document -> Site Header
 - Line before Site Header -> Content

In that case, you'd have to `Tab, Select (Skiplink #1, go to site header), Shift+Tab (Back one), Select (Skiplink #2, go to content)`. The only way to reach the second one was to go through the full header, and then skip over the much smaller "site header".

Reply:
> The customization of the skip destination was primarily to avoid having to change all of the existing themes structure, while removing the header dependancies.

Makes sense 👍🏻 

> In that case, the links didn't make sense as they were.

Agreed, I'd call that a bug you fixed 🙂